Classes/Outlet/AbstractOutlet.php

```diff
@@ -146,8 +176,97 @@ public function produce()
     {
         $data = $this->data;
         foreach ($this->pipesOut as $pipe) {
+            if ($pipe instanceof ViewAwarePipeInterface) {
+                $pipe->setView($this->view);
+            }
             $pipe->conduct($data);
         }
+
         return $data;
     }
+
+    /**
+     * @return ViewInterface
+     */
+    public function getView()
+    {
+        return $this->view;
+    }
+
+    /**
+     * @param ViewInterface $view
+     * @return OutletInterface
+     */
+    public function setView($view)
+    {
+        $this->view = $view;
+
+        return $this;
+    }
+
+    /**
+     * @return OutletArgument[]
+     */
+    public function getArguments()
+    {
+        return $this->arguments;
+    }
+
+    /**
+     * @param OutletArgument[] $arguments
+     */
+    public function setArguments(array $arguments)
+    {
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * @param OutletArgument $argument
+     */
+    public function addArgument(OutletArgument $argument)
+    {
+        $this->arguments[] = $argument;
+    }
+
+    /**
+     * Validate given $data based on configured argument validations
+     *
+     * @param $data
+     * @return Result
+     */
+    public function validate($data)
+    {
+        $this->validationResults = new Result();
+        foreach ($this->arguments as $argument) {
+            $argument->setValue(isset($data[$argument->getName()]) ? $data[$argument->getName()] : null);
```

Comment:
?? is only supported since 7, right? composer.json currently specifies 5.5+, so the ull coalescing operator is not usable, imho